### PR TITLE
Only display contact info for minors to those cleared for it.

### PR DESCRIPTION
**Pull request by cinnion**
- Add a new VIEW_MINOR_PII restricted permission to determine if a user
  is cleared to view the contact info of minors.
- Add new config value, minor_pii_config, which defines the default age of
  majority along with criteria for locales which do not match the default.
- Add a new method to MedusaPermissions, canViewMinorPii(), which handles
  the logic of deciding whether or not the contact info should be visible.
- Wrap the contact info section in the blade with a conditional based on
  whether they are themselves, or should be able to view based on the new
  permission logic.
- Add VIEW_MINOR_PII to the restricted permissions in the .env.example
  file.

### CODE REVIEW

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        //
```
Comment (dweinerATL) [inline database/migrations/2021_02_27_175238_view_minor_pii_perm.php:50]: @cinnion Add code to the down method that will remove the permission and add an entry to the audit trail

File: database/migrations/2021_02_27_175400_view_minor_pii_data.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+	MedusaConfig::remove('minor_pii_config');
+    }
```
Comment (dweinerATL) [inline database/migrations/2021_02_27_175400_view_minor_pii_data.php:49]: @cinnion Add audit trail to the up and down methods

**Review by dweinerATL — CHANGES_REQUESTED** [review]

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php (hunk 1 above)
Comment (cinnion) [reply database/migrations/2021_02_27_175238_view_minor_pii_perm.php]: Pushed.

File: database/migrations/2021_02_27_175400_view_minor_pii_data.php (hunk 2 above)
Comment (cinnion) [reply database/migrations/2021_02_27_175400_view_minor_pii_data.php:69]: Pushed.

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            );
+            App\Permission::create(['name' => $perm, 'description' => $desc]);
+        }
+
+        // Add the new permission to the Dir MDOC and 5SL
+
+        foreach (['RMN-1094-12', 'RMN-4637-17'] as $member_id) {
+            $user = \App\User::where('member_id', $member_id)->first();
+            $user->updatePerms(['VIEW_MINOR_PII']);
+            $user->save();
```
Comment (dweinerATL) [inline database/migrations/2021_02_27_175238_view_minor_pii_perm.php:39]: I'm sorry @cinnion, I missed this the last time.  We also need an audit trail entry for adding the permission to these users

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public function down()
+    {
+        $delPerms = [
+            'VIEW_MINOR_PII' => 'Can view the PII of minors',
+        ];
+
+        foreach ($delPerms as $perm => $desc) {
+
+            foreach (\App\User::where('permissions', $perm)->get() as $user) {
+                $user->deletePerm($perm);
```
Comment (dweinerATL) [inline database/migrations/2021_02_27_175238_view_minor_pii_perm.php:57]: @cinnion and here too

**Review by dweinerATL — CHANGES_REQUESTED** [review]

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php (hunk 3 above)
Comment (cinnion) [reply database/migrations/2021_02_27_175238_view_minor_pii_perm.php:39]: updatePerms does its own audit trail creation. Do you want additional audit trail records?

File: database/migrations/2021_02_27_175238_view_minor_pii_perm.php (hunk 4 above)
Comment (cinnion) [reply database/migrations/2021_02_27_175238_view_minor_pii_perm.php:57]: Same as updatePerms().

**Pull request merged**